### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by drainingsun** [robertklep/node-metrohash#2]
It seems that node-metrohash performs very poorly compared to other Node.js fast hash implementations. 6-7 times worse than some other hash methods. You can test it yourself with this gist: https://gist.github.com/drainingsun/a2974657164e661722a14d214228b89c

Any idea why this could be happening? The test was also performed without `toString("hex")` method with no significant gains.

**Comment by robertklep** [comment]
This is most likely due to my lacking C++ skills  🙄

I'll try and see if I can export the Metrohash library functions from C++ to Node with only a thin layer, instead of (as it's currently implemented) as a class that gets instantiated for each hashing operation (although I'm pretty sure that if you only use `.hash()`, you can reuse the same instance for each hash calculation).

**Comment by drainingsun** [comment]
You are right on the class instantiation part! There was around 40% increase in ops/sec after removing the instantiation from the loop. Hopefully you will find a way to make it even faster!

**Comment by robertklep** [comment]
@drainingsun I pushed v1.5.1 to Github (not published to NPM yet), but it doesn't look like I can get it to perform a whole lot better (EDIT: see below).

If you'd like to try, you can install it from GH directly:
```
$ npm i robertklep/node-metrohash#v1.5.1
```

**EDIT**: okay, I tracked down the source of the slowness: it's converting the array containing the calculated digest to a Node `Buffer`. I'll try and see if I can speed things up.

**EDIT**: I managed to turn the loser into a winner (well, at least for 1024 bytes, I didn't bother to test any other sizes):

```
1024 bytes
  12 tests completed.

  sha1                   x   399,744 ops/sec ±1.58% (83 runs sampled)
  farmHash-hash64        x 1,095,349 ops/sec ±1.32% (87 runs sampled)
  farmHash-fingerprint64 x 1,135,497 ops/sec ±0.74% (92 runs sampled)
  metroHash-64           x 3,594,674 ops/sec ±0.77% (92 runs sampled)
  metroHash-128          x 3,335,276 ops/sec ±1.11% (89 runs sampled)
  murmurHash-64          x 2,177,016 ops/sec ±0.93% (90 runs sampled)
  murmurHash-64x86       x 1,835,682 ops/sec ±0.67% (94 runs sampled)
  murmurHash-64x64       x 2,164,777 ops/sec ±0.93% (90 runs sampled)
  murmurHash-128         x 1,887,468 ops/sec ±0.75% (94 runs sampled)
  murmurHash-128x86      x 1,462,499 ops/sec ±0.79% (93 runs sampled)
  murmurHash-128x64      x 1,902,176 ops/sec ±0.71% (91 runs sampled)
  xxHash-64              x 2,171,373 ops/sec ±0.61% (92 runs sampled)

Fastest is: metroHash-64
Slowest is: sha1
```

As you can see, it also removes the large standard deviation.

The change is that the standalone functions now return a hex string, instead of a buffer, which seems fine to me. You can run a full test against `master`:

```
$ npm i robertklep/node-metrohash
```

I feel like I should probably convert the class-based hashers to returning HEX strings as well, since that seems to be the fastest way of getting the hash to the JS-side of Node, and it's trivial to convert such a string to a Buffer, if so required.

**Comment by drainingsun** [comment]
Just tested it myself. New implementation is a beast! Up to 60% faster than murmurHash. Now that's a performance one can awe at. Thank you! 

P.S. Agree on the HEX strings part.

**Comment by robertklep** [comment]
@drainingsun I pushed a separate branch ([v2.0](https://github.com/robertklep/node-metrohash/tree/v2.0)), if you'd like to test. I removed the `.hash()` methods altogether, because even declaring them as just a (static) class method seemed to add some overhead. Instead, the standalone functions now take an optional `seed` argument.

If it works okay, I'll merge and publish as `v2.0.0` to the NPM repo :D
